Automattic/wp-calypso
# Add support for Jetpack products to <PurchasesListing /> on my-plan page

So far the new `<PurchasesListing />` component on `/plans/my-plan` page displayed information only about the plan the site is on.

With this PR the purchases listing presents information about the Jetpack products as well.

**Please note that the `<PlanIcon />` component doesn't support Jetpack Backup yet. This will be addressed in a subsequent PR and shouldn't block this change set.**

**Before**
![Screenshot 2019-11-19 at 18 15 28](https://user-images.githubusercontent.com/478735/69169714-1ee9e180-0af9-11ea-9a85-e5c46af42214.png)

**After** 
![Screenshot 2019-11-19 at 18 14 50](https://user-images.githubusercontent.com/478735/69169717-201b0e80-0af9-11ea-82b1-8d48505ac659.png)

#### Changes proposed in this Pull Request

* Add support for Jetpack products to `<PurchasesListing />` on `/plans/my-plan` page
* Clean up `<PurchasesListing />` component
* Add Jetpack Backup tag line copy to constants file
* Add necessary utility functions to `product-values` library

#### Testing instructions



* Go to http://calypso.localhost:3000/plans/my-plan/
* Select a Jetpack site that has a Daily backup product
* Confirm that the product is displayed in the "My Products" section
* Using SA change the expiration date to less than 30 days from now and confirm that the expiration information is now red
* Upgrade to Real-Time and confirm that the correct title and tag line is displayed

Fixes n/a
